### ISSUE TITLE
Non-deterministic initialisation of function pointers is broken

### ISSUE THREAD

**Issue by xbauch** [diffblue/cbmc#4937]
```main.c
#include <assert.h>

typedef int (*other_function_type)(int n);

void foo(other_function_type other_function)
{
  assert(other_function(4) > 5);
}
```

CBMC version: 5.12 (cbmc-5.12-d8598f8-559-gfaf21d043-dirty)
Operating system: Ubuntu 18.04
Exact command line resulting in the issue: `$ cbmc main.c --function foo`
What behaviour did you expect: VERIFICATION FAILED
What happened instead: `assignment to 'symbol' not handled`

**Comment by xbauch** [comment]
```
__CPROVER__start
{
  static signed int tmp(signed int);
  other_function_type other_function;
  if(NONDET(__CPROVER_bool))
    other_function = ((other_function_type)NULL);
  else
  {
    other_function = tmp;
    tmp = NONDET(signed int (signed int));
  }
  foo(other_function);
}
```
This is what the generated entry-point looks like.

**Comment by tautschnig** [comment]
I believe this has been fixed in #4940. @xbauch Would you please confirm and close this issue? And perhaps review #5209?

**Comment by xbauch** [comment]
Fixed by #4940.
